src/database/votePost.js

```diff
@@ -0,0 +1,13 @@
+import { db } from "database";
+
+export const voteForOptA = (postObj) => {
+  db.collection("posts")
+    .doc(postObj.id)
+    .update({ option_a: postObj.option_a + 1 });
+};
+
+export const voteForOptB = (postObj) => {
+  db.collection("posts")
+    .doc(postObj.id)
+    .update({ option_b: postObj.option_b + 1 });
+};
```

Comment:
This is code duplication. The only difference in these two codes is `option_a` and `option_b`. You could easily accept them as parameters when you call the function, so that the function remains generic and you don't repeat yourself in code

Reply:
I was trying to do that but I couldn't figure out how to make a Obj key a variable

Reply:
Object keys can be variables like the following

```
const myObj = {
  [varKeyName]: "some value",
  [otherVarKeyName]: "some other value",
}
```